File: python/valuecell/server/api/schemas/holding_lifecycle.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class HoldingLifecycleItemData(BaseModel):
    holding_id: int
    ticker: str
    display_name: str
    lifecycle_stage: str
    action: str
    confidence: int
    summary: str
    theme_name: str | None = None
    role_label: str | None = None
    trend_quality: str | None = None
    tradeability_state: str | None = None
    expectation_state: str | None = None
    observation_window: str
    invalid_conditions: list[str] = Field(default_factory=list)
    risk_controls: list[str] = Field(default_factory=list)
    profit_protection_view: str | None = None
    position_hint: str
    has_active_alerts: bool
    decision_context_available: bool


class HoldingLifecycleSummaryData(BaseModel):
    total_count: int
    need_attention_count: int
    major_hold_count: int
    high_risk_count: int
    active_alert_count: int


class HoldingLifecycleOverviewData(BaseModel):
    generated_at: datetime
    available: bool
    empty_message: str | None = None
    items: list[HoldingLifecycleItemData] = Field(default_factory=list)
    count: int
    summary: HoldingLifecycleSummaryData


class HoldingLifecycleDetailData(BaseModel):
    generated_at: datetime
    available: bool
    empty_message: str | None = None
    item: HoldingLifecycleItemData

File: frontend/src/types/holding-lifecycle.ts
export type HoldingLifecycleStage =
  | "建仓观察期"
  | "主升持有期"
  | "分歧确认期"
  | "退潮减仓期"
  | "破逻辑退出期";

export type HoldingLifecycleItem = {
  holding_id: number;
  ticker: string;
  display_name: string;
  lifecycle_stage: HoldingLifecycleStage;
  action: string;
  confidence: number;
  summary: string;
  theme_name: string | null;
  role_label: string | null;
  trend_quality: string | null;
  tradeability_state: string | null;
  expectation_state: string | null;
  observation_window: string;
  invalid_conditions: string[];
  risk_controls: string[];
  profit_protection_view: string | null;
  position_hint: string;
  has_active_alerts: boolean;
  decision_context_available: boolean;
};

export type HoldingLifecycleOverview = {
  generated_at: string;
  available: boolean;
  empty_message: string | null;
  items: HoldingLifecycleItem[];
  count: number;
  summary: {
    total_count: number;
    need_attention_count: number;
    major_hold_count: number;
    high_risk_count: number;
    active_alert_count: number;
  };
};

export type HoldingLifecycleDetail = {
  generated_at: string;
  available: boolean;
  empty_message: string | null;
  item: HoldingLifecycleItem;
};
